Mlem/Extensions/JSONDecoder+Default.swift

```diff
@@ -11,35 +11,33 @@ extension JSONDecoder {
     static var defaultDecoder: JSONDecoder {
         let decoder = JSONDecoder()
         decoder.keyDecodingStrategy = .convertFromSnakeCase
-        
+
         let formatter = DateFormatter()
         
+        formatter.timeZone = .gmt
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        
+        let formats = [
+            "yyyy-MM-dd'T'HH:mm:ss.SSSSSS",
+            "yyyy-MM-dd'T'HH:mm:ss.SSSSZ",
+            "yyyy-MM-dd'T'HH:mm:ss.SSSS",
+            "yyyy-MM-dd'T'HH:mm:ss",
+            "yyyy-MM-dd HH:mm:ss.SSSSSS",
+            "yyyy-MM-dd HH:mm:ss",
+            "yyyy-MM-dd"
+        ]
```

Comment:
FWIW there's only a few kinds of date formats that Lemmy is expected to return, but this should work

Separately; if it hits the last one, is it possible it'll only parse the first part of the string, and disregard the rest? Or should it match exactly?

Reply:
I think It should match exactly unfortunately, which is why I added it as a last resort

Reply:
yeah, the formatter is very specific so the last would only map to dates in that exact format, I agree that I don't think we've ever seen that (or should see it), with it being last in the list it likely won't hit it often, other than before failing.

Happy to get this merged if you are @ShadowJonathan?

Reply:
If it maps in an exact format, it should be fine, approved.